src/System.Private.CoreLib/shared/System/Text/UTF8Encoding.cs

```diff
@@ -170,22 +181,78 @@ public override unsafe int GetByteCount(string chars)
         public override unsafe int GetByteCount(char* chars, int count)
         {
             // Validate Parameters
+
             if (chars == null)
```

Comment:
`chars is null`, while you are here?

Reply:
Can you explain why that's better?

Reply:
It probably doesn't matter for pointer types as much, but for regular types where you can override `==`, it gives a fast null check that doesn't call into any user-defined code. Updating it here would just make it more consistent with updates we are making elsewhere.

Reply:
@ahsonkhan see also https://github.com/dotnet/coreclr/pull/21765

Reply:
The "_x_ is null" pattern does not compile if _x_ is a pointer type.